azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -0,0 +1,468 @@
+package azurerm
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2017-09-30/containerservice"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmKubernetesCluster() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmKubernetesClusterCreate,
+		Read:   resourceArmKubernetesClusterRead,
+		Update: resourceArmKubernetesClusterCreate,
+		Delete: resourceArmKubernetesClusterDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"dns_prefix": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"kubernetes_version": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"linux_profile": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"admin_username": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"ssh_key": {
+							Type:     schema.TypeList,
+							Required: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"key_data": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"agent_pool_profile": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+
+						"count": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							Default:      1,
+							ValidateFunc: validation.IntBetween(1, 100),
```

Comment:
It looks like the API docs say between 1 and 100 for the count here but the Azure Portal says it can only be between 1 and 50 so we should reflect that here. 
![image](https://user-images.githubusercontent.com/6515518/35878693-e064f908-0b2d-11e8-9b26-f413c5fddb2c.png)

Reply:
Yes, looks like there are differences between docs, cli and portal. Changed max to 50 for now.